src/actions/historyActions.js

```diff
@@ -478,13 +481,25 @@ export const restoreTransactionHistoryAction = () => {
  * For the key based wallet it uses ethplorer as a data provider
  * For smart wallets data will be fetched through the Archanova SDK
  */
-export const fetchTransactionsHistoryAction = () => {
+export const fetchTransactionsHistoryAction = (forAllAccounts?: boolean) => {
   return async (dispatch: Dispatch, getState: GetState) => {
     const {
       accounts: { data: accounts },
       appSettings: { data: { blockchainNetwork } = {} },
+      featureFlags: {
+        data: {
+          BITCOIN_ENABLED: bitcoinFeatureEnabled,
+        },
+      },
     } = getState();
 
+    if (forAllAccounts) {
+      await dispatch(restoreTransactionHistoryAction());
+      await dispatch(fetchSmartWalletTransactionsAction());
+      if (bitcoinFeatureEnabled) dispatch(fetchBTCTransactionsHistoryAction());
```

Comment:
BTC fetch is handled in the older PR, the inner action checks if there are any addresses generated and handles transactions accordingly https://github.com/pillarwallet/pillarwallet/pull/1979/files#diff-7ea48032f831c00d653dddebf080d034R487

Reply:
Hm, this is some other case. It fetches history for all accounts.
The one you refer to, fetches history of BTC and of active acc. On Home feed we need it to be fetched for BTC, SW and KW at the same time.
I've also using the same `fetchBTCTransactionsHistoryAction` method to fetch BTC history that is used in the method you refer to. So is there an issue? Am I missing something? :)